zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -576,17 +701,59 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             call.rw_counter_end_of_reversion,
             call.is_persistent,
         )?;
-        self.sufficient_gas_left
-            .assign(region, offset, F::from(tx.gas() - step.gas_cost))?;
+
+        let init_code_gas_cost = if tx.is_create() {
+            // Calculate gas cost of init code for EIP-3860.
+            (tx.call_data.len() as u64 + 31) / 32 * eth_types::evm_types::INIT_CODE_WORD_GAS
```

Comment:
The length of calldata, you could get from the return value of `self.tx_call_data_word_length.assign(...)`

Reply:
a nitpick, you could import `INIT_CODE_WORD_GAS` in L31, `use eth_types::{....}`, then the code here looks cleaner

Reply:
Good point. Updated.